mysql-test/include/log_encrypt_3.inc

```diff
@@ -0,0 +1,206 @@
+# InnoDB transparent encryption on redo log.
+# This test case will test basic encryption support features.
+--source include/no_valgrind_without_big.inc
+--source include/have_innodb.inc
+--source include/not_embedded.inc
+--source include/have_innodb_max_16k.inc
+
+let $old_innodb_file_per_table = `SELECT @@innodb_file_per_table`;
+let $old_innodb_redo_log_encrypt = `SELECT @@innodb_redo_log_encrypt`;
+CREATE DATABASE tde_db;
+USE tde_db;
+
+# Server is not started with keyring plugin and run innodb_redo_log_encrypt=MASTER_KEY/ROTATED_KEY
+# Expected Result : Variable is ignored.
+# bug : expect warning or error
+eval SET GLOBAL innodb_redo_log_encrypt = $redo_log_encrypt_mode;
+SHOW WARNINGS;
+--let $wait_condition=SELECT @@global.innodb_redo_log_encrypt = 0
+--source include/wait_condition.inc
+CREATE TABLE tde_db.t4 (a BIGINT PRIMARY KEY, b LONGBLOB) ENGINE=InnoDB;
+INSERT INTO t4 (a, b) VALUES (1, REPEAT('a', 6*512*512));
+SELECT a,LEFT(b,10) FROM tde_db.t4;
+
+# Restart server
+--source include/restart_mysqld.inc
+
+SELECT a,LEFT(b,10) FROM tde_db.t4;
+DROP TABLE tde_db.t4;
+DROP DATABASE tde_db;
+CREATE DATABASE tde_db;
+USE tde_db;
+
+# Start server with --early-plugin-load and --innodb_redo_log_encrypt=$redo_log_encrypt_mode .
+# Explicitly run , INSTALL PLUGIN keyring_file SONAME 'keyring_file.so'
+# Expected : plugin already installed
+--echo # Starting server with keyring plugin
+let $restart_parameters = restart: --early-plugin-load="keyring_file=$KEYRING_PLUGIN" --loose-keyring_file_data=$MYSQL_TMP_DIR/my_key_log3 $KEYRING_PLUGIN_OPT --innodb_redo_log_encrypt=$redo_log_encrypt_mode;
+--replace_result $MYSQL_TMP_DIR MYSQL_TMP_DIR $KEYRING_PLUGIN_OPT --plugin-dir=KEYRING_PLUGIN_PATH $KEYRING_PLUGIN keyring_file.so
+--source include/restart_mysqld_no_echo.inc
+
+SELECT @@global.innodb_redo_log_encrypt;
+
+--replace_regex /\.dll/.so/
+--error ER_UDF_EXISTS
+eval INSTALL PLUGIN keyring_file SONAME '$KEYRING_PLUGIN';
+UNINSTALL PLUGIN keyring_file;
+--echo # Starting server with keyring plugin
+let $restart_parameters = restart: --early-plugin-load="keyring_file=$KEYRING_PLUGIN" --loose-keyring_file_data=$MYSQL_TMP_DIR/my_key_log3 $KEYRING_PLUGIN_OPT --innodb_redo_log_encrypt=$redo_log_encrypt_mode;
+--replace_result $MYSQL_TMP_DIR MYSQL_TMP_DIR $KEYRING_PLUGIN_OPT --plugin-dir=KEYRING_PLUGIN_PATH $KEYRING_PLUGIN keyring_file.so
+--source include/restart_mysqld_no_echo.inc
+
+SET GLOBAL innodb_redo_log_encrypt = 0;
+SELECT @@global.innodb_redo_log_encrypt;
```

Comment:
Not sure what these SELECTs past SET are supposed to test (OK to fix post-merge or never)